@cachedout Here's the stack trace:
```
[DEBUG   ] Traceback (most recent call last):
  File "c:\salt-dev\salt\salt\states\file.py", line 1586, in managed
    follow_symlinks)
  File "c:\salt-dev\salt\salt\modules\file.py", line 3756, in manage_file
    ret, _ = check_perms(name, ret, user, group, mode, follow_symlinks)
  File "c:\salt-dev\salt\salt\modules\file.py", line 3243, in check_perms
    if isinstance(orig_comment, six.string_types):
NameError: global name 'six' is not defined
```

This is a clean install. I did a `git reset --hard` and a `git clean -fxd` on the salt clone. I removed python from the system and removed the `C:\python27` folder before reinstalling. I do this everytime I do a build to make sure it's all clean.

--- a/salt/modules/file.py
+++ b/salt/modules/file.py
@@ -33,10 +33,7 @@ import glob
 import mmap
 
 # pylint: disable=import-error,no-name-in-module,redefined-builtin
-# This string_types import needs to be here for windows as six.string_types
-# does not work from this module... This only applies to the 2015.5 branch.
-# Do not merge forward
-from salt.ext.six import string_types
+import salt.ext.six as six
 from salt.ext.six.moves import range, reduce, zip
 from salt.ext.six.moves.urllib.parse import urlparse as _urlparse
 # pylint: enable=import-error,no-name-in-module,redefined-builtin
@@ -2888,7 +2885,7 @@ def source_list(source, source_hash, saltenv):
                 elif single_src.startswith('/') and os.path.exists(single_src):
                     ret = (single_src, single_hash)
                     break
-            elif isinstance(single, string_types):
+            elif isinstance(single, six.string_types):
                 path, senv = salt.utils.url.parse(single)
                 if not senv:
                     senv = saltenv
@@ -3243,7 +3240,7 @@ def check_perms(name, ret, user, group, mode, follow_symlinks=False):
         elif 'cgroup' in perms and user != '':
             ret['changes']['group'] = group
 
-    if isinstance(orig_comment, string_types):
+    if isinstance(orig_comment, six.string_types):
         if orig_comment:
             ret['comment'].insert(0, orig_comment)
         ret['comment'] = '; '.join(ret['comment'])
@@ -3500,7 +3497,7 @@ def get_diff(
 
     ret = ''
 
-    if isinstance(env, string_types):
+    if isinstance(env, six.string_types):
         salt.utils.warn_until(
             'Boron',
             'Passing a salt environment should be done using \'saltenv\' not '

--- a/salt/modules/win_file.py
+++ b/salt/modules/win_file.py
@@ -29,8 +29,8 @@ import sys  # do not remove, used in imported file.py functions
 import fileinput  # do not remove, used in imported file.py functions
 import fnmatch  # do not remove, used in imported file.py functions
 import mmap  # do not remove, used in imported file.py functions
-import salt.ext.six as six  # pylint: disable=import-error,no-name-in-module
 # do not remove, used in imported file.py functions
+import salt.ext.six as six  # pylint: disable=import-error,no-name-in-module
 from salt.ext.six.moves.urllib.parse import urlparse as _urlparse  # pylint: disable=import-error,no-name-in-module
 import salt.utils.atomicfile  # do not remove, used in imported file.py functions
 from salt.exceptions import CommandExecutionError, SaltInvocationError

--- a/salt/modules/win_file.py
+++ b/salt/modules/win_file.py
@@ -29,7 +29,7 @@ import sys  # do not remove, used in imported file.py functions
 import fileinput  # do not remove, used in imported file.py functions
 import fnmatch  # do not remove, used in imported file.py functions
 import mmap  # do not remove, used in imported file.py functions
-from salt.ext.six import string_types  # do not remove, used in imported file.py functions
+import salt.ext.six as six  # pylint: disable=import-error,no-name-in-module
 # do not remove, used in imported file.py functions
 from salt.ext.six.moves.urllib.parse import urlparse as _urlparse  # pylint: disable=import-error,no-name-in-module
 import salt.utils.atomicfile  # do not remove, used in imported file.py functions

